frontend/packages/core/src/NPS/feedback.tsx

```diff
@@ -0,0 +1,281 @@
+  ratingLabels: [
+    {
+      emoji: 1,
+      label: "bad",
+    },
+    {
+      emoji: 2,
+      label: "ok",
+    },
+    {
+      emoji: 3,
+      label: "great",
+    },
+  ],
+};
+
+/** Styling */
+
+const Text = styled.span({
+  fontWeight: "bold",
+  color: "#0D1030",
+  fontSize: "14px",
+});
+
+const StyledGrid = styled(MuiGrid)({
+  borderRadius: "8px",
+  padding: "20px",
+});
+
+const StyledAlert = styled(Alert)({
+  margin: "32px",
+  alignItems: "center",
+});
+
+/** Components */
+
+/**
+ * EmojiRatings component which will take an array of emojis and given ratings and create IconButtons with them and return them on selection
+ *
+ * @param ratings given array of ratings to display
+ * @param setRating function which will return the given selected rating
+ * @param size allows overriding of a given size, "large" is the only appropriate value
+ * @returns rendered EmojiRatings component
+ */
+export const EmojiRatings = ({ ratings = [], setRating }) => {
+  const [selectedRating, selectRating] = useState<Rating>(null);
+
+  const StyledIconButton = styled(IconButton)<{
+    selected: boolean;
+  }>(
+    {
+      "&:hover": {
+        opacity: 1,
+      },
+    },
+    props => ({
+      opacity: props.selected ? 1 : 0.5,
+    })
+  );
+
+  const select = (rating: Rating) => {
+    selectRating(rating);
+    setRating(rating);
+  };
+
+  // Will convert a given integer to a typed enum key if necessary
+  const getKey = (map, val) => Object.keys(map).find(key => map[key] === val);
+
+  return (
+    <>
+      {ratings.map((rating: Rating) => {
+        const { label } = rating;
+        let { emoji } = rating;
+
+        if (isInteger(emoji)) {
+          emoji = getKey(IClutch.feedback.v1.EmojiRating, emoji);
+        }
+
+        return (
+          <Tooltip key={label} title={capitalize(label)}>
+            <StyledIconButton
+              key={`rating-${emoji}`}
+              variant="neutral"
+              size="medium"
+              selected={selectedRating?.label === label}
+              onClick={() => select(rating)}
+            >
+              <Emoji type={emoji} size="large" />
+            </StyledIconButton>
+          </Tooltip>
+        );
+      })}
+    </>
+  );
+};
+
+/**
+ * NPS feedback component which is the base for both Wizard and Anytime
+ * Will fetch given survey options from the server based on the given origin
+ * Then display a feedback component based on the given emoji ratings
+ *
+ * @param opts Available feedback options
+ * @returns NPSFeedback component
+ */
+const NPSFeedback = (opts: FeedbackOptions = { origin: "ORIGIN_UNSPECIFIED" }) => {
+  const [hasSubmit, setSubmit] = useState<boolean>(false);
+  const [selected, setSelected] = useState<Rating>(null);
+  const [feedback, setFeedback] = useState<string>("");
+  const [error, setError] = useState<boolean>(false);
+  const [survey, setSurvey] = useState<IClutch.feedback.v1.ISurvey>({});
+  const requestId = uuid();
```

Comment:
current the requestId is changing on each submitFeedack request so we won't be able to update the user's feedback submission

doing something like this will fix that
```
const [requestId, setrequestId] = useState<string>("");

// set requestId just 1x
....
React.useEffect(() => {
    setrequestId(uuid())
  }, []);

```

Reply:
Weird, I had tested that and it didn't change at all. Must have broken during refactoring.

Good catch, will adjust.